src/ape_test/provider.py

```diff
@@ -133,8 +134,15 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         receipt = self.get_receipt(
             txn_hash.hex(), required_confirmations=txn.required_confirmations or 0
         )
-        if txn.gas_limit is not None and receipt.ran_out_of_gas:
-            raise OutOfGasError()
+
+        if receipt.failed:
```

Comment:
Can transactions fail for reasons that cause it to not be idempotent? That is to say, something like a network connection error or similar would it could fail the first time but pass the second time if sent again?

Reply:
If we get a receipt at all at this point, the transaction was published.

Reply:
Oh right, was confusing this with a preemptive failure like gas estimation

Also was forgetting this is `call()` and not a second sending of the transaction 

I think we're good then 👍